src/editor/components/modals/file-modal.jsx

```diff
@@ -0,0 +1,230 @@
+import React from "react";
+import AppBar from "@material-ui/core/AppBar";
+import Toolbar from "@material-ui/core/Toolbar";
+import Typography from "@material-ui/core/Typography";
+import styled from "react-emotion";
+import { connect } from "react-redux";
+
+import DeleteModal from "../../../server/components/delete-modal";
+import THEME from "../../../shared/theme";
+import { ModalContainer } from "./modal-container";
+import { List, ListItem } from "../../../server/components/list";
+import {
+  ContainedButton,
+  OutlineButton
+} from "../../../shared/components/buttons";
+import { updateAppMessages } from "../../actions/actions";
+import {
+  saveFileToServer
+} from "../../../shared/utils/file-operations";
+
+const gridGap = "5px";
+
+const ModalContent = styled.div`
+  padding: 20px;
+`;
+
+const AddFileButton = styled(ContainedButton)`
+  align-self: start;
+  margin: 0;
+`;
+
+const FileList = styled(List)`
+  margin: 20px 0;
+  overflow-y: auto;
+`;
+
+const FileListItem = styled(ListItem)`
+  align-items: center;
+  display: grid;
+  grid-gap: ${gridGap};
+  grid-template-areas: "title actions";
+  height: unset;
+`;
+
+const FileName = styled.span`
+  font-weight: bold;
+  grid-area: title;
+`;
+
+const FilePath = styled.input`
+  position: absolute;
+  top: -10000px;
+  left: -10000px;
+`;
+
+const FileActions = styled.div`
+  display: flex;
+  grid-area: actions;
+  justify-content: flex-end;
+`;
+
+const FileAction = styled(OutlineButton)`
+  &:last-child {
+    margin-right: 0;
+  }
+`;
+
+class FileModalUnconnected extends React.Component {
+  constructor(props) {
+    super(props);
+    this.fileInputRef = React.createRef();
+    this.notebookInfo = JSON.parse(
+      document.getElementById("notebookInfo").innerHTML
+    );
+    this.notebookId = this.notebookInfo.notebook_id;
+    this.state = {
+      deleteModalVisible: false,
+      files: this.notebookInfo.files
+    };
+  }
+
+  onFileReceipt = e => {
+    const file = e.target.files[0];
+    const fileReader = new FileReader();
+
+    fileReader.readAsArrayBuffer(file);
+
+    fileReader.onload = () => {
+      saveFileToServer(this.notebookId, fileReader.result, file.name)
+        .then(savedFile => {
+          this.setState(state => {
+            state.files.push(savedFile);
+            return state;
+          });
+          this.props.updateAppMessages({
```

Comment:
these don't 100% work properly, I think. While it definitely shows a message in the snackbar, it does not show anything in the console. I recognize this is a failing we have with moving from dangerously set content for app messages in the console -> boilerplate messages, but we'll have to figure out something here before this merges.

Otherwise they all appear blank in the console:

<img width="1085" alt="Screen Shot 2019-04-23 at 3 11 28 PM" src="https://user-images.githubusercontent.com/95735/56619349-1e787f00-65da-11e9-8686-ada28a42a3d5.png">

Reply:
Huh. I didn't notice that! Are messages supposed to be simul-printed to the console?

Reply:
Is there an example of an `updateAppMessages` message that is printed to the console correctly?